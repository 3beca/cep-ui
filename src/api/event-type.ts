import {
    fetchData,
    RequestInfo,
    ResponseData,
    ErrorAPI,
    ResponseEmptyData
} from './fetch-data';
import {
    BASE_URL,
    EVENT_TYPE_URL
} from './config';
export * from './fetch-data';

export type EventType = {
    name: string;
    id: string;
    url: string;
    createdAt: string;
    updatedAt: string;
};
export type EventTypeList = {
    results: EventType[];
    next?: string | undefined;
    prev?: string | undefined;
};
export const getEventTypeList = (config: RequestInfo) => async (page: number = 1, size: number = 10): Promise<ResponseData<EventTypeList>|ErrorAPI> => {
    const url = `${config.url}${EVENT_TYPE_URL}/?page=${page}&pageSize=${size}`;
    return await fetchData<undefined, EventTypeList>({...config, url, method: 'GET'});
};
export const deleteEventType = (config: RequestInfo) => async (eventTypeIds: string|string[]): Promise<(ResponseEmptyData|ErrorAPI)[]> => {
    const eventTypesArray = (Array.isArray(eventTypeIds)) ? eventTypeIds : [eventTypeIds];
    if(!eventTypeIds || eventTypesArray.length < 1) {
        return [{
            status: 500,
            error: 'Missing EventType id',
            message: 'eventTypeIds is an invalid EventTypeId value or array'
        }];
    }
    const requests = [];
    for (const eventTypeId of eventTypesArray) {
        const url = `${config.url}${EVENT_TYPE_URL}/${eventTypeId}`;
        requests.push(fetchData<undefined, undefined>({...config, url, method: 'DELETE'}));
    }
    return Promise.all(requests);
};
export type EventTypeApi = {
    getEventTypeList(page?: number, size?: number): Promise<ResponseData<EventTypeList>|ErrorAPI>;
    deleteEventType(eventTypeIds: string|string[]): Promise<(ResponseEmptyData|ErrorAPI)[]>;
};
export const EventTypeApi = (): EventTypeApi => {
    const config: RequestInfo = {
        url: BASE_URL,
        method: 'GET',
        headers: {}
    };
    return {
        getEventTypeList: getEventTypeList(config),
        deleteEventType: deleteEventType(config)
    };
};
