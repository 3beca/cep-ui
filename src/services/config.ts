export const BASE_URL = window._env_.BASE_URL;
export const EVENT_TYPE_URL = '/admin/event-types';