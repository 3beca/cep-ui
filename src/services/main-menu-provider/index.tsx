import * as React from 'react';
import { NOOP } from '../../utils';
export type MainMenuToggleContextValue = {
    toggle: () => void;
    close: () => void;
    open: () => void;
};
const MainMenuStateContext = React.createContext(false);
const MainMenuToggleContext = React.createContext<MainMenuToggleContextValue>({
    toggle: NOOP,
    close: NOOP,
    open: NOOP
});

export const MainMenuProvider: React.FC<{}> = props => {
    const [isOpen, setOpen] = React.useState(false);
    const toggle = React.useCallback(() => setOpen(isOpen => !isOpen), []);
    const open = React.useCallback(() => setOpen(true), []);
    const close = React.useCallback(() => setOpen(false), []);
    const utils = React.useMemo(
        () => ({
            toggle,
            open,
            close
        }),
        [toggle, open, close]
    );

    return (
        <MainMenuToggleContext.Provider value={utils}>
            <MainMenuStateContext.Provider value={isOpen} {...props} />
        </MainMenuToggleContext.Provider>
    );
};

export const useMainMenuState = () => {
    return React.useContext(MainMenuStateContext);
};

export const useMainMenuToggle = () => {
    return React.useContext(MainMenuToggleContext);
};
