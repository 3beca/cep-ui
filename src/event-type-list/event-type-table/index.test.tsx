import React from 'react';
import { render } from '@testing-library/react';
import EventTypeTable from './index';
import { EventTypeList } from '../../services/event-type';

const generateListWith = (many: number = 5, next = false, prev = false) => {
    const list: EventTypeList = {
        results: Array(many).map((_, idx) => ({
            id: idx + '',
            name: 'Elemento ' + idx,
            url: 'http://cep/elemento' + idx,
            createdAt: '2020-01-01T10:10:10.123Z',
            updatedAt: '2020-01-01T10:10:10.123Z'
        }))
    };
    if (prev) list.prev = 'http://cep/?page=prev';
    if (next) list.next = 'http://cep/?page=next';
    return list;
};

test('EventTypeTable mount without crash', () => {
    const { getByText } = render(<EventTypeTable />);
    const linkElement = getByText(/Table of Event Types/i);
    expect(linkElement).toBeInTheDocument();
});

test('EventTypeTable snapshot without data', () => {
    const { container } = render(<EventTypeTable />);
    expect(container).toMatchSnapshot();
});

test('EventTypeTable snapshot with data and pagination', () => {
    const { container } = render(
        <EventTypeTable
            eventTypeList={generateListWith(5, true, true)}
            isLoading={false}
            page={2}
            size={5}
        />
    );
    expect(container).toMatchSnapshot();
});

test('EventTypeTable snapshot while loading with previous data and pagination', () => {
    const { container } = render(
        <EventTypeTable
            eventTypeList={generateListWith(5, true, true)}
            isLoading={true}
            page={2}
            size={5}
        />
    );
    expect(container).toMatchSnapshot();
});

test('EventTypeTable snapshot while loading without data', () => {
    const { container } = render(
        <EventTypeTable
            eventTypeList={generateListWith(0, false, false)}
            isLoading={true}
            page={1}
            size={5}
        />
    );
    expect(container).toMatchSnapshot();
});

test('EventTypeTable should render 5 items in page 2and pagination prev and next enabled', () => {
    const onChangePage = jest.fn();
    const {getByText, getByTitle, getByLabelText, rerender} = render(
        <EventTypeTable
            eventTypeList={undefined}
            isLoading={false}
            page={1}
            size={5}
            onChangePage={onChangePage}
        />
    );
    const prevButton = getByTitle('Previous page') as HTMLButtonElement;
    const nextButton = getByTitle('Next page') as HTMLButtonElement;
    const rowsPerPage = getByLabelText('Rows per page:');

    expect(prevButton.disabled).toBe(true);
    expect(nextButton.disabled).toBe(true);
    expect(rowsPerPage).toHaveTextContent('5');
    //expect(getByText('6-10 of 100')).toHaveTextContent('2');

    // Rerender with new data
    rerender(
        <EventTypeTable
            eventTypeList={generateListWith(5, true, true)}
            isLoading={false}
            page={1}
            size={5}
            onChangePage={onChangePage}
        />
    );

    expect(prevButton.disabled).toBe(true);
    expect(nextButton.disabled).toBe(false);
    expect(rowsPerPage).toHaveTextContent('5');
    expect(getByText('6-10 of 100')).toHaveTextContent('2');
});