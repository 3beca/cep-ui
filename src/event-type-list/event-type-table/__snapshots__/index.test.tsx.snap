// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventTypeTable snapshot while loading with previous data and pagination 1`] = `
<div>
  <div
    class="makeStyles-root-430"
  >
    <span>
      Table of Event Types
    </span>
    <div
      class="MuiTableContainer-root"
    >
      <table
        class="MuiTable-root"
      >
        <thead
          class="MuiTableHead-root makeStyles-head-431"
        >
          <tr
            class="MuiTableRow-root MuiTableRow-head"
          >
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-paddingCheckbox"
              scope="col"
            >
              <span
                aria-disabled="true"
                class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-461 MuiCheckbox-root makeStyles-mainCheck-433 PrivateSwitchBase-disabled-463 Mui-disabled Mui-disabled Mui-disabled"
                tabindex="-1"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <input
                    class="PrivateSwitchBase-input-464"
                    data-indeterminate="false"
                    disabled=""
                    type="checkbox"
                    value=""
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                    />
                  </svg>
                </span>
              </span>
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
              scope="col"
            >
              <p
                class="MuiTypography-root makeStyles-headText-432 MuiTypography-body1"
              >
                Event Type Name
              </p>
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
              scope="col"
            >
              <p
                class="MuiTypography-root makeStyles-headText-432 MuiTypography-body1"
              >
                Last updated
              </p>
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
              scope="col"
            >
              <p
                class="MuiTypography-root makeStyles-headText-432 MuiTypography-body1"
              >
                Created At
              </p>
            </th>
          </tr>
        </thead>
        <tbody
          class="MuiTableBody-root"
        >
          <tr
            class="MuiTableRow-root"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-paddingCheckbox"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-461 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <input
                    class="PrivateSwitchBase-input-464"
                    data-indeterminate="false"
                    type="checkbox"
                    value=""
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </span>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
            >
              Dos
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              2020-1-1 11:10:10
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              2020-1-1 12:11:11
            </td>
          </tr>
          <tr
            class="MuiTableRow-root"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body"
              colspan="4"
              style="background-color: pink;"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter"
              >
                Loading...
              </p>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div
      class="MuiTablePagination-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiTablePagination-toolbar MuiToolbar-gutters"
      >
        <div
          class="MuiTablePagination-spacer"
        />
        <p
          class="MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit"
        >
          Rows per page:
        </p>
        <div
          class="MuiInputBase-root MuiTablePagination-input MuiTablePagination-selectRoot"
        >
          <div
            aria-haspopup="listbox"
            aria-label="Rows per page:"
            class="MuiSelect-root MuiSelect-select MuiTablePagination-select MuiSelect-selectMenu MuiInputBase-input"
            role="button"
            tabindex="0"
          >
            5
          </div>
          <input
            type="hidden"
            value="5"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSelect-icon MuiTablePagination-selectIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
        <p
          class="MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit"
        >
          6-6 of 6
        </p>
        <div
          class="MuiTablePagination-actions"
        >
          <button
            aria-label="Previous page"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
            tabindex="0"
            title="Previous page"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            aria-label="Next page"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
            disabled=""
            tabindex="-1"
            title="Next page"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`EventTypeTable snapshot while loading without data 1`] = `
<div>
  <div
    class="makeStyles-root-573"
  >
    <span>
      Table of Event Types
    </span>
    <div
      class="MuiTableContainer-root"
    >
      <table
        class="MuiTable-root"
      >
        <thead
          class="MuiTableHead-root makeStyles-head-574"
        >
          <tr
            class="MuiTableRow-root MuiTableRow-head"
          >
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-paddingCheckbox"
              scope="col"
            >
              <span
                aria-disabled="true"
                class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-604 MuiCheckbox-root makeStyles-mainCheck-576 PrivateSwitchBase-disabled-606 Mui-disabled Mui-disabled Mui-disabled"
                tabindex="-1"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <input
                    class="PrivateSwitchBase-input-607"
                    data-indeterminate="false"
                    disabled=""
                    type="checkbox"
                    value=""
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                    />
                  </svg>
                </span>
              </span>
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
              scope="col"
            >
              <p
                class="MuiTypography-root makeStyles-headText-575 MuiTypography-body1"
              >
                Event Type Name
              </p>
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
              scope="col"
            >
              <p
                class="MuiTypography-root makeStyles-headText-575 MuiTypography-body1"
              >
                Last updated
              </p>
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
              scope="col"
            >
              <p
                class="MuiTypography-root makeStyles-headText-575 MuiTypography-body1"
              >
                Created At
              </p>
            </th>
          </tr>
        </thead>
        <tbody
          class="MuiTableBody-root"
        >
          <tr
            class="MuiTableRow-root"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-paddingCheckbox"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-604 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <input
                    class="PrivateSwitchBase-input-607"
                    data-indeterminate="false"
                    type="checkbox"
                    value=""
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </span>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
            >
              Dos
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              2020-1-1 11:10:10
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              2020-1-1 12:11:11
            </td>
          </tr>
          <tr
            class="MuiTableRow-root"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body"
              colspan="4"
              style="background-color: pink;"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter"
              >
                Loading...
              </p>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div
      class="MuiTablePagination-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiTablePagination-toolbar MuiToolbar-gutters"
      >
        <div
          class="MuiTablePagination-spacer"
        />
        <p
          class="MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit"
        >
          Rows per page:
        </p>
        <div
          class="MuiInputBase-root MuiTablePagination-input MuiTablePagination-selectRoot"
        >
          <div
            aria-haspopup="listbox"
            aria-label="Rows per page:"
            class="MuiSelect-root MuiSelect-select MuiTablePagination-select MuiSelect-selectMenu MuiInputBase-input"
            role="button"
            tabindex="0"
          >
            5
          </div>
          <input
            type="hidden"
            value="5"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSelect-icon MuiTablePagination-selectIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
        <p
          class="MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit"
        >
          1-5 of 5
        </p>
        <div
          class="MuiTablePagination-actions"
        >
          <button
            aria-label="Previous page"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
            disabled=""
            tabindex="-1"
            title="Previous page"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                />
              </svg>
            </span>
          </button>
          <button
            aria-label="Next page"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
            disabled=""
            tabindex="-1"
            title="Next page"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`EventTypeTable snapshot with data and pagination 1`] = `
<div>
  <div
    class="makeStyles-root-287"
  >
    <span>
      Table of Event Types
    </span>
    <div
      class="MuiTableContainer-root"
    >
      <table
        class="MuiTable-root"
      >
        <thead
          class="MuiTableHead-root makeStyles-head-288"
        >
          <tr
            class="MuiTableRow-root MuiTableRow-head"
          >
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-paddingCheckbox"
              scope="col"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-318 MuiCheckbox-root makeStyles-mainCheck-290"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <input
                    class="PrivateSwitchBase-input-321"
                    data-indeterminate="false"
                    type="checkbox"
                    value=""
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </span>
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
              scope="col"
            >
              <p
                class="MuiTypography-root makeStyles-headText-289 MuiTypography-body1"
              >
                Event Type Name
              </p>
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
              scope="col"
            >
              <p
                class="MuiTypography-root makeStyles-headText-289 MuiTypography-body1"
              >
                Last updated
              </p>
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
              scope="col"
            >
              <p
                class="MuiTypography-root makeStyles-headText-289 MuiTypography-body1"
              >
                Created At
              </p>
            </th>
          </tr>
        </thead>
        <tbody
          class="MuiTableBody-root"
        >
          <tr
            class="MuiTableRow-root"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-paddingCheckbox"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-318 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <input
                    class="PrivateSwitchBase-input-321"
                    data-indeterminate="false"
                    type="checkbox"
                    value=""
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </span>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
            >
              Dos
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              2020-1-1 11:10:10
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              2020-1-1 12:11:11
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div
      class="MuiTablePagination-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiTablePagination-toolbar MuiToolbar-gutters"
      >
        <div
          class="MuiTablePagination-spacer"
        />
        <p
          class="MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit"
        >
          Rows per page:
        </p>
        <div
          class="MuiInputBase-root MuiTablePagination-input MuiTablePagination-selectRoot"
        >
          <div
            aria-haspopup="listbox"
            aria-label="Rows per page:"
            class="MuiSelect-root MuiSelect-select MuiTablePagination-select MuiSelect-selectMenu MuiInputBase-input"
            role="button"
            tabindex="0"
          >
            5
          </div>
          <input
            type="hidden"
            value="5"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSelect-icon MuiTablePagination-selectIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
        <p
          class="MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit"
        >
          6-6 of 6
        </p>
        <div
          class="MuiTablePagination-actions"
        >
          <button
            aria-label="Previous page"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
            tabindex="0"
            title="Previous page"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            aria-label="Next page"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
            disabled=""
            tabindex="-1"
            title="Next page"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`EventTypeTable snapshot without data 1`] = `
<div>
  <div
    class="makeStyles-root-144"
  >
    <span>
      Table of Event Types
    </span>
    <div
      class="MuiTableContainer-root"
    >
      <table
        class="MuiTable-root"
      >
        <thead
          class="MuiTableHead-root makeStyles-head-145"
        >
          <tr
            class="MuiTableRow-root MuiTableRow-head"
          >
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-paddingCheckbox"
              scope="col"
            >
              <span
                aria-disabled="true"
                class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-175 MuiCheckbox-root makeStyles-mainCheck-147 PrivateSwitchBase-disabled-177 Mui-disabled Mui-disabled Mui-disabled"
                tabindex="-1"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <input
                    class="PrivateSwitchBase-input-178"
                    data-indeterminate="false"
                    disabled=""
                    type="checkbox"
                    value=""
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                    />
                  </svg>
                </span>
              </span>
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
              scope="col"
            >
              <p
                class="MuiTypography-root makeStyles-headText-146 MuiTypography-body1"
              >
                Event Type Name
              </p>
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
              scope="col"
            >
              <p
                class="MuiTypography-root makeStyles-headText-146 MuiTypography-body1"
              >
                Last updated
              </p>
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
              scope="col"
            >
              <p
                class="MuiTypography-root makeStyles-headText-146 MuiTypography-body1"
              >
                Created At
              </p>
            </th>
          </tr>
        </thead>
        <tbody
          class="MuiTableBody-root"
        >
          <tr
            class="MuiTableRow-root"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-paddingCheckbox"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-175 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <input
                    class="PrivateSwitchBase-input-178"
                    data-indeterminate="false"
                    type="checkbox"
                    value=""
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </span>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
            >
              Dos
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              2020-1-1 11:10:10
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              2020-1-1 12:11:11
            </td>
          </tr>
          <tr
            class="MuiTableRow-root"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body"
              colspan="4"
              style="background-color: pink;"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter"
              >
                Loading...
              </p>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div
      class="MuiTablePagination-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiTablePagination-toolbar MuiToolbar-gutters"
      >
        <div
          class="MuiTablePagination-spacer"
        />
        <p
          class="MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit"
        >
          Rows per page:
        </p>
        <div
          class="MuiInputBase-root MuiTablePagination-input MuiTablePagination-selectRoot"
        >
          <div
            aria-haspopup="listbox"
            aria-label="Rows per page:"
            class="MuiSelect-root MuiSelect-select MuiTablePagination-select MuiSelect-selectMenu MuiInputBase-input"
            role="button"
            tabindex="0"
          >
            10
          </div>
          <input
            type="hidden"
            value="10"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSelect-icon MuiTablePagination-selectIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
        <p
          class="MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit"
        >
          1-10 of 10
        </p>
        <div
          class="MuiTablePagination-actions"
        >
          <button
            aria-label="Previous page"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
            disabled=""
            tabindex="-1"
            title="Previous page"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                />
              </svg>
            </span>
          </button>
          <button
            aria-label="Next page"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
            disabled=""
            tabindex="-1"
            title="Next page"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
